File: apps/pos/native-android/app/src/main/java/com/desktopkitchen/pos/models/Payment.kt
package com.desktopkitchen.pos.models

import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class PaymentIntent(
    val client_secret: String,
    val payment_intent_id: String,
    val amount: Double
)

@JsonClass(generateAdapter = true)
data class PaymentStatus(
    val status: String,
    val payment_intent_id: String? = null,
    val amount: Double? = null
)

@JsonClass(generateAdapter = true)
data class CreatePaymentIntentRequest(
    val order_id: Int,
    val tip: Double? = null
)

@JsonClass(generateAdapter = true)
data class ConfirmPaymentRequest(
    val order_id: Int,
    val payment_intent_id: String
)

@JsonClass(generateAdapter = true)
data class CashPaymentRequest(
    val order_id: Int,
    val tip: Double? = null,
    val amount_received: Double? = null
)

File: apps/pos/native-windows/DesktopKitchenPOS/Models/Payment.cs
using System.Text.Json.Serialization;

namespace DesktopKitchenPOS.Models;

public record PaymentIntent
{
    [JsonPropertyName("client_secret")]
    public string ClientSecret { get; init; } = "";

    [JsonPropertyName("payment_intent_id")]
    public string PaymentIntentId { get; init; } = "";

    [JsonPropertyName("amount")]
    public double Amount { get; init; }
}

public record PaymentStatus
{
    [JsonPropertyName("status")]
    public string Status { get; init; } = "";

    [JsonPropertyName("payment_intent_id")]
    public string? PaymentIntentId { get; init; }

    [JsonPropertyName("amount")]
    public double? Amount { get; init; }
}

public record CreatePaymentIntentRequest
{
    [JsonPropertyName("order_id")]
    public int OrderId { get; init; }

    [JsonPropertyName("tip")]
    public double? Tip { get; init; }
}

public record ConfirmPaymentRequest
{
    [JsonPropertyName("order_id")]
    public int OrderId { get; init; }

    [JsonPropertyName("payment_intent_id")]
    public string PaymentIntentId { get; init; } = "";
}

public record CashPaymentRequest
{
    [JsonPropertyName("order_id")]
    public int OrderId { get; init; }

    [JsonPropertyName("tip")]
    public double? Tip { get; init; }

    [JsonPropertyName("amount_received")]
    public double? AmountReceived { get; init; }
}
